### ISSUE TITLE
duckdb run the tpcds query no.85 failed, and no.64 out of memory

### ISSUE THREAD

**Issue by l1t1** [duckdb/duckdb#2907]
#### What happens?
duckdb run one of the tpcds query failed

```
--64
Error: near line 1: Out of Memory Error: could not allocate block of 262144 bytes
Database is launched in in-memory mode and no temporary directory is specified.
Unused blocks cannot be offloaded to disk.

Launch the database with a persistent storage back-end

--85
Run Time: real 1147.346 user 1146.872552 sys 0.015600
Error: near line 2: INTERRUPT Error: Interrupted!
```

#### To Reproduce

I copied the sql from 

https://codeload.github.com/cwida/tpcds-result-reproduction/zip/refs/heads/master

extract to d:/tpcds/app

and copied the data from

https://github.com/Altinity/tpc-ds/tree/master/data

extract to d:/tpcds/data

the file customer_001.dat need to be converted to utf8

I wrote the import command in d:/import_tpcds.txt

then run .read d:/tpcds_sql_bat.txt

It stoped after the above output.

		

#### Environment (please complete the following information):
 - OS: [e.g. iOS]windows7 x64
 - DuckDB Version: [e.g. 22]0.30
 - DuckDB Client: [e.g. Python]CLI

#### Before Submitting

- [ ] **Have you tried this on the latest `master` branch?**
* **Python**: `pip install duckdb --upgrade --pre`
* **R**: `install.packages("https://github.com/duckdb/duckdb/releases/download/master-builds/duckdb_r_src.tar.gz", repos = NULL)`
* **Other Platforms**: You can find binaries [here](https://github.com/duckdb/duckdb/releases/tag/master-builds) or compile from source.

- [ ] **Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?**

**Comment by l1t1** [comment]
[import_tpcds.txt](https://github.com/duckdb/duckdb/files/7844040/import_tpcds.txt)
[tpcds_sql_bat.txt](https://github.com/duckdb/duckdb/files/7844042/tpcds_sql_bat.txt)

**Comment by l1t1** [comment]
[64.sql.txt](https://github.com/duckdb/duckdb/files/7844538/64.sql.txt)
[85.sql.txt](https://github.com/duckdb/duckdb/files/7844537/85.sql.txt)

**Comment by Mytherin** [comment]
This has been fixed

**Comment by ngould** [comment]
@Mytherin Do you have a PR reference for the fix? I just saw something similar using `0.5.1`, and am wondering whether I should file a bug report or if this covers it.

**Comment by Mytherin** [comment]
The problem was a suboptimal join order leading to an exploding join - the fix was improving the join order (see https://github.com/duckdb/duckdb/pull/4274).
